aws/resource_aws_ecs_service.go

```diff
@@ -843,6 +916,11 @@ func resourceAwsEcsServiceUpdate(d *schema.ResourceData, meta interface{}) error
 		input.NetworkConfiguration = expandEcsNetworkConfiguration(d.Get("network_configuration").([]interface{}))
 	}
 
+	if d.HasChange("capacity_provider_strategy") {
+		updateService = true
+		input.CapacityProviderStrategy = expandEcsCapacityProviderStrategy(d.Get("capacity_provider_strategy").([]interface{}))
```

Comment:
This likely would panic as the attribute type is currently `TypeSet`, it might be good to verify this via an update test.

```suggestion
		input.CapacityProviderStrategy = expandEcsCapacityProviderStrategy(d.Get("capacity_provider_strategy").(*schema.Set).List())
```